### PR TITLE
Mb/get csv

### PR DESCRIPTION
**Pull request by matt6deg**
Created `QueryWriteCSV()` method that sends output from a database query directly into a CSV file on the local filesystem.

I _think_ I put `QueryWriteCSV()` methods into every DBer interface, but not certain about that.

Also had some issues where it wanted to import from the original version rather than the forked version. I _think_ I eliminated those, at least got to the point where it worked locally for me, but again, not certain.

### CODE REVIEW

File: dbToJson.go [hunk 1]
```diff
@@ -128,13 +128,15 @@ func getJSONValue(pval *interface{}) string {
 // over the bytes.Buffer
 var hex = "0123456789abcdef"
 
-func encodeByteSlice(byteSlice []byte) string {
+func encodeByteSlice(byteSlice []byte, useQuotes bool) string {
```
Comment (robarchibald) [inline dbToJson.go]: remove

File: dbToJson_test.go [hunk 2]
```diff
@@ -128,16 +128,20 @@ func TestGetJsonValue(t *testing.T) {
 }
 
 func TestEncodeByteSlice(t *testing.T) {
-	if actual := encodeByteSlice([]byte{}); actual != "null" {
+	if actual := encodeByteSlice([]byte{}, true); actual != "null" {
```
Comment (robarchibald) [inline dbToJson_test.go]: remove

File: dbToCSV.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	switch v := (*pval).(type) {
+	case nil:
+		return ""
+	case bool:
+		if v {
+			return "true"
+		}
+		return "false"
+	case []byte:
+		return encodeByteSlice(v, false)
```
Comment (robarchibald) [inline dbToCSV.go]: This is no different than case default

File: dbToCSV.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	case time.Time:
+		timeFormat := "2006-01-02 15:04:05.999"
+		if options["dateOnly"] {
```
Comment (robarchibald) [inline dbToCSV.go]: Change to struct

**Comment by matt6deg** [comment]
Ready for re-review, all issues resolved to the best of my knowledge.

**Pull request merged**